### ISSUE TITLE
How to force the mapper to get the same number of cameras and images as the input?

### ISSUE THREAD

**Issue by haoyi-duan** [colmap/colmap#2832]
Hi, When I do automatic reconstruction based on the input images, I find that sometimes the solutions (the output cameras.txt and images.txt files) are missing images/cameras during the mapping. I wonder if it is possible to force the mapper to generate the same number of cameras and images as the number of the input image?

**Comment by tsattler** [comment]
> I wonder if it is possible to force the mapper to generate the same number of cameras and images as the number of the input image?

No, there is no way. If colmap cannot register an image, it is not included in the output.

**Comment by jaco001** [comment]
To the some degre you can 'fake' data that COLMAP will process, Like cropping, sharpening etc. but again why?
COLMAP is fair. Help it to get good data not make it confuse. Make better dataset first.

**Comment by ahojnnes** [comment]
Closing: COLMAP cannot force registration of all images; by design.
